gh-108951: document addition of TaskGroup.cancel() (#149031)
python/cpython: Doc/whatsnew/3.15.rst

  This can be :ref:`controlled using environment variables <using-on-controlling-color>`.
  (Contributed by Stan Ulbrych in :gh:`148981`.)

asyncio
-------

> * Added :meth:`~asyncio.TaskGroup.cancel` to allow early termination of a task group --

Reviewer: Add the missing blank line.
```suggestion

* Added :meth:`~asyncio.TaskGroup.cancel` to allow early termination of a task group --
```

  This can be :ref:`controlled using environment variables <using-on-controlling-color>`.
  (Contributed by Stan Ulbrych in :gh:`148981`.)

asyncio
-------

* Added :meth:`~asyncio.TaskGroup.cancel` to allow early termination of a task group --

>   e.g. when the goal of the tasks has been achieved or their services no longer needed.

Reviewer 2: ```suggestion
  e.g. when the goal of the tasks has been achieved or their services are no longer needed.
```

Reviewer 3: I've included this in my suggestion as well. Thanks! (my suggestion was mainly: rewmoving the "e.g." as per the devguide and have two blank lines before and after each section)

>   This can be :ref:`controlled using environment variables <using-on-controlling-color>`.
>   (Contributed by Stan Ulbrych in :gh:`148981`.)
> 
> asyncio
> -------
> 
> * Added :meth:`~asyncio.TaskGroup.cancel` to allow early termination of a task group --
>   e.g. when the goal of the tasks has been achieved or their services no longer needed.
>   Previously this would involve unintuitive boilerplate such as an extra task raising
>   a custom exception which is then suppressed as it exits TaskGroup.
>   (Contributed by John Belmonte in :gh:`127214`.)
> 
> base64

Reviewer 3: ```suggestion
  (Contributed by Stan Ulbrych in :gh:`148981`.)

asyncio
-------

* Added :meth:`~asyncio.TaskGroup.cancel` to allow early termination of a task group,
  for instance, when the goal of the tasks has been achieved or their services
  are no longer needed. Previously this would involve unintuitive boilerplate such
  as an extra task raising a custom exception which is then suppressed as it
  exits the task group.
  (Contributed by John Belmonte in :gh:`127214`.)

base64
```

Reviewer: This is just flat out wrong. `TaskGroup.cancel()` cancels the entire group, not just a single task.

Reviewer 3: Oh I removed the word "group". Thanks for finding it. I just wanted to reflow the paragraph.

Reviewer 2: @picnixz Did you fix the suggestion? I can't find the missing 'group'.

Reviewer 3: Yes, I edited my suggestion. Let me check it again.

  (Contributed by Stan Ulbrych in :gh:`148981`.)

asyncio
-------

* Added :meth:`~asyncio.TaskGroup.cancel` to allow early termination of a task group,
  for instance, when the goal of the tasks has been achieved or their services
  are no longer needed. Previously this would involve unintuitive boilerplate such
  as an extra task raising a custom exception which is then suppressed as it
  exits the task group.

>   (Contributed by John Belmonte in :gh:`127214`.)

Reviewer 3: ```suggestion
* Added :meth:`TaskGroup.cancel <asyncio.TaskGroup.cancel>` to allow early
  termination of a task group, for instance, when the goal of the tasks has
  been achieved or their services are no longer needed.
  Previously this would involve unintuitive boilerplate such as an extra task
  raising a custom exception which is then suppressed as it exits the task group.
  (Contributed by John Belmonte in :gh:`127214`.)
```
